File: tennis-intelligence-platform/api/schemas/player_profile.py
"""
schemas/player_profile.py — Pydantic response models for the Player Profile page.
"""

from __future__ import annotations

from pydantic import BaseModel


class PlayerSearchResult(BaseModel):
    player_id: str
    player_name: str


class SurfaceStat(BaseModel):
    matches: int
    wins: int
    win_pct: float | None


class GrandSlamStats(BaseModel):
    matches: int
    wins: int
    win_pct: float | None


class RecentFormEntry(BaseModel):
    match_id: str
    date: str | None
    opponent: str
    won: bool
    surface: str
    tournament: str


class EloTimelinePoint(BaseModel):
    match_id: str
    date: str | None
    elo: float | None
    surface_elo: float | None


class PlayerProfileResponse(BaseModel):
    player_id: str
    player_name: str
    current_elo: float | None
    peak_elo: float | None
    career_matches: int
    career_wins: int
    career_losses: int
    career_win_pct: float | None
    surface_stats: dict[str, SurfaceStat]
    grand_slam_stats: GrandSlamStats
    recent_form: list[RecentFormEntry]
    elo_timeline: list[EloTimelinePoint]


class HeadToHeadMatch(BaseModel):
    match_id: str
    date: str | None
    tournament: str
    surface: str
    winner_id: str
    score: str | None


class HeadToHeadResponse(BaseModel):
    player_id_a: str
    player_id_b: str
    a_wins: int
    b_wins: int
    matches: list[HeadToHeadMatch]

File: tennis-intelligence-platform-web/src/types/playerProfile.ts
// types/playerProfile.ts — matches api/schemas/player_profile.py exactly.

export interface PlayerSearchResult {
  player_id: string;
  player_name: string;
}

export interface SurfaceStat {
  matches: number;
  wins: number;
  win_pct: number | null;
}

export interface GrandSlamStats {
  matches: number;
  wins: number;
  win_pct: number | null;
}

export interface RecentFormEntry {
  match_id: string;
  date: string | null;
  opponent: string;
  won: boolean;
  surface: string;
  tournament: string;
}

export interface EloTimelinePoint {
  match_id: string;
  date: string | null;
  elo: number | null;
  surface_elo: number | null;
}

export interface PlayerProfileResponse {
  player_id: string;
  player_name: string;
  current_elo: number | null;
  peak_elo: number | null;
  career_matches: number;
  career_wins: number;
  career_losses: number;
  career_win_pct: number | null;
  surface_stats: Record<string, SurfaceStat>;
  grand_slam_stats: GrandSlamStats;
  recent_form: RecentFormEntry[];
  elo_timeline: EloTimelinePoint[];
}

export interface HeadToHeadMatch {
  match_id: string;
  date: string | null;
  tournament: string;
  surface: string;
  winner_id: string;
  score: string | null;
}

export interface HeadToHeadResponse {
  player_id_a: string;
  player_id_b: string;
  a_wins: number;
  b_wins: number;
  matches: HeadToHeadMatch[];
}
